packages/amplify-graphql-model-transformer/rds-lambda/handler.ts

```diff
@@ -24,10 +24,34 @@ export const run = async (event): Promise<any> => {
     adapter = await getDBAdapter(config);
   }
   const debugMode = process.env.DEBUG_MODE === 'true';
-  const result = await adapter.executeRequest(event, debugMode);
-  return result;
+  try {
+    return await adapter.executeRequest(event, debugMode);
+  } catch (e) {
+    if (isRetryableError(e)) {
+      return await retryWithRefreshedCredentials(event, debugMode)
+    }
+    throw e;
+  }
+};
+
+const retryWithRefreshedCredentials = async (event, debugMode): Promise<any> => {
+  try {
+    const config = await getDBConfig();
+    adapter = await getDBAdapter(config);
+    return await adapter.executeRequest(event, debugMode);
+  } catch (err) {
+    adapter = null;
+    throw err;
+  }
 };
 
+const isRetryableError = (error): boolean => {
+  const postgresRetryableError = error.code === '28P01' && error.routine === 'auth_failed';
+  const mysqlRetruableError = error.errno === '1045' && error.code === 'ER_ACCESS_DENIED_ERROR';
+
+  return postgresRetryableError || mysqlRetruableError;
```

Comment:
nit 

```suggestion
  const mysqlRetryableError = error.errno === '1045' && error.code === 'ER_ACCESS_DENIED_ERROR';

  return postgresRetryableError || mysqlRetryableError;
```

Reply:
Could you also add links to the sources of each of these error numbers & codes, so if we need to extend them in future we have a good starting point?